diff -ur cops_104/docs/makefile cops-1.0.5rc1/docs/makefile
--- cops_104/docs/makefile	1992-03-10 13:57:18.000000000 -0500
+++ cops-1.0.5rc1/docs/makefile	2023-05-19 21:58:00.000000000 -0400
@@ -17,14 +17,14 @@
 #
 # Where the programs are....
 #
-NROFF=/usr/bin/nroff
-RM=/bin/rm -f
+NROFF=/bin/nroff
+RM=/bin/rm
 
 # make all
 all:	$(DOCS) $(MAN)
 
 clean:
-	$(RM) $(DOCS) $(MAN)
+	$(RM) -f $(DOCS) $(MAN)
 
 # 'roff out those docs
 COPS.report.ms: COPS.report
diff -ur cops_104/makefile cops-1.0.5rc1/makefile
--- cops_104/makefile	1992-03-10 13:57:49.000000000 -0500
+++ cops-1.0.5rc1/makefile	2023-05-19 22:13:10.000000000 -0400
@@ -4,10 +4,11 @@
 #	make all	    -- makes everything
 #	make install	    -- puts things in their place
 #	make <program_name> -- make a given program
-INSTALL_DIR= sun
+#INSTALL_DIR= sun
+INSTALL_DIR= bin
 
-EXECUTABLE = home.chk user.chk pass.chk is_writable crc crc_check \
-	     addto clearfiles filewriters members tilde is_able
+EXECUTABLE = home.chk user.chk is_writable crc crc_check \
+	     addto clearfiles filewriters members tilde is_able  pass.chk 
 C_SRC      = home.chk.c user.chk.c is_able.c pass.c is_something.c \
 	     addto.c clearfiles.c filewriters.c members.c tilde.c \
 	     crc.c crc_check.c
@@ -26,7 +27,8 @@
 
 #  Certain systems need to uncomment this to compile the pass.chk; Xenix,
 # some SysV:
-# BRAINDEADFLAGS = -lcrypt
+# Needed for RHEL8
+BRAINDEADFLAGS = -lcrypt
 #
 # systems without rindex need to uncomment this:
 # CRC_FLAG=-Dstrrchr=rindex
@@ -66,43 +68,43 @@
 
 # make the programs
 addto: src/addto.c
-	$(CC) $(CFLAGS) -o addto src/addto.c
+	$(CC) $(CFLAGS) -o $@ $?
 
 clearfiles: src/clearfiles.c
-	$(CC) $(CFLAGS) -o clearfiles src/clearfiles.c
+	$(CC) $(CFLAGS) -o $@ $?
 
 filewriters: src/filewriters.c
-	$(CC) $(CFLAGS) -o filewriters src/filewriters.c
+	$(CC) $(CFLAGS) -o $@ $?
 
 members: src/members.c
-	$(CC) $(CFLAGS) -o members src/members.c
+	$(CC) $(CFLAGS) -o $@ $?
 
 home.chk: src/home.chk.c
-	$(CC) $(CFLAGS) -o home.chk src/home.chk.c
+	$(CC) $(CFLAGS) -o $@ $?
 
 user.chk: src/user.chk.c
-	$(CC) $(CFLAGS) -o user.chk src/user.chk.c
+	$(CC) $(CFLAGS) -o $@ $?
 
 is_able: src/is_able.c
-	$(CC) $(CFLAGS) -o is_able src/is_able.c
+	$(CC) $(CFLAGS) -o $@ $?
 
 is_writable: src/is_something.c
-	$(CC) $(CFLAGS) -DWRITABLE -o is_writable src/is_something.c
+	$(CC) $(CFLAGS) -DWRITABLE -o $@ $?
 
 #   If fast crypt will work, comment the first CC line, uncomment
 # the next two:
 pass.chk: src/pass.c
-	$(CC) $(CFLAGS) -o pass.chk src/pass.c $(BRAINDEADFLAGS)
-# 	$(CC) $(CFLAGS) -Dcrypt=fcrypt -DFCRYPT -o pass.chk src/pass.c \
+	$(CC) $(CFLAGS) $(BRAINDEADFLAGS) -o $@ $?
+# 	$(CC) $(CFLAGS) -Dcrypt=fcrypt -DFCRYPT -o $@ \
 # 	src/crack-fcrypt.c src/crack-lib.c $(BRAINDEADFLAGS)
 
 tilde: src/tilde.c
-	$(CC) $(CFLAGS) -o tilde src/tilde.c
+	$(CC) $(CFLAGS) -o $@ $?
 
 crc: src/crc.c
-	$(CC) $(CFLAGS) -o crc src/crc.c $(SEQFLAGS)
+	$(CC) $(CFLAGS) -o $@ $(SEQFLAGS) $?
 
 crc_check: src/crc_check.c
-	$(CC) $(CFLAGS) $(CRC_FLAG) -o crc_check src/crc_check.c $(SEQFLAGS)
+	$(CC) $(CFLAGS) $(CRC_FLAG) -o $@$(SEQFLAGS) $?
 
 # the end
diff -ur cops_104/reconfig cops-1.0.5rc1/reconfig
--- cops_104/reconfig	1992-03-10 13:58:01.000000000 -0500
+++ cops-1.0.5rc1/reconfig	2023-05-19 19:18:35.000000000 -0400
@@ -205,5 +205,5 @@
 	$MV $i.new $i
 	# finally, make sure everything is back to executable status
 	$CHMOD u+x $i
-
 done
+chmod a-x docs/makefile
diff -ur cops_104/src/pass.c cops-1.0.5rc1/src/pass.c
--- cops_104/src/pass.c	1992-03-10 13:58:13.000000000 -0500
+++ cops-1.0.5rc1/src/pass.c	2023-05-19 22:09:36.000000000 -0400
@@ -2,12 +2,16 @@
 #include <pwd.h>
 #include <ctype.h>
 
+/* Added for RHEL 8 */
+#include <crypt.h>
+#include <string.h>
+
 /* C2 stuff by Ole H. Nielsen */
 #ifdef C2
 #include <sys/label.h>
 #include <sys/audit.h>
 #include <pwdadj.h>
-#endif C2
+#endif
 
 /* number of words the dictionary can suck up */
 #define ARB_CONST	32000
@@ -282,7 +286,7 @@
 #ifdef C2
     struct passwd_adjunct *pwdadj;
     struct passwd_adjunct *getpwanam();
-#endif C2
+#endif
     register char	*cp, *cp2;
     struct passwd	*pwd;
     struct passwd	*getpwent();
@@ -362,7 +366,7 @@
 			/* Substitute the C2 secure password */
 			pwd->pw_passwd = pwdadj->pwa_passwd;
 	}
-#endif C2
+#endif
 
 	if (*pwd->pw_passwd == '\0') {
 	    if (chknulls) {
@@ -722,7 +726,9 @@
  * NULL if not found
  */
 
+#ifndef NULL
 #define	NULL	0
+#endif
 
 char *
 my_index(sp, c)
